### ISSUE TITLE
AbstractTimestampLongConverter.append() performance issue

### ISSUE THREAD

**Issue by maxim-ponomarev** [OpenHFT/Chronicle-Wire#623]
`AbstractTimestampLongConverter.append()` throws, catches and swallows `UnsupportedTemporalTypeException` on every invocation, which results in a performance penalty.
Reproduced on Java 8.
Not reproduced on Java 17.

```
java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: OffsetSeconds
	at java.time.LocalDate.get0(LocalDate.java:680)
	at java.time.LocalDate.getLong(LocalDate.java:659)
	at java.time.LocalDateTime.getLong(LocalDateTime.java:720)
	at java.time.format.DateTimePrintContext.getValue(DateTimePrintContext.java:298)
	at java.time.format.DateTimeFormatterBuilder$OffsetIdPrinterParser.format(DateTimeFormatterBuilder.java:3346)
	at java.time.format.DateTimeFormatterBuilder$CompositePrinterParser.format(DateTimeFormatterBuilder.java:2190)
	at java.time.format.DateTimeFormatterBuilder$CompositePrinterParser.format(DateTimeFormatterBuilder.java:2190)
	at java.time.format.DateTimeFormatter.formatTo(DateTimeFormatter.java:1746)
	at net.openhft.chronicle.wire.AbstractTimestampLongConverter.append(AbstractTimestampLongConverter.java:128)
	at net.openhft.chronicle.wire.AbstractTimestampLongConverter.append(AbstractTimestampLongConverter.java:137)
	at net.openhft.chronicle.wire.AbstractTimestampLongConverterPerfTest$Task.run(AbstractTimestampLongConverterPerfTest.java:45)
	at net.openhft.chronicle.jlbh.JLBH.warmup(JLBH.java:286)
	at net.openhft.chronicle.jlbh.JLBH.start(JLBH.java:165)
	at net.openhft.chronicle.wire.AbstractTimestampLongConverterPerfTest.test(AbstractTimestampLongConverterPerfTest.java:25)
```

**Comment by maxim-ponomarev** [comment]
@nicktindall 
It looks like the problem is caused by `appendOffsetId` [here](https://github.com/OpenHFT/Chronicle-Wire/blob/72c00d21151252a183882f85c5336362744bfe9e/src/main/java/net/openhft/chronicle/wire/AbstractTimestampLongConverter.java#L107).
Do you know anything about it?

**Comment by nicktindall** [comment]
Wow great pickup, removing that really seems to speed things up.

The reason it's there is because we use the same `DateTimeFormatter` for parsing and formatting, and we need to be lenient for ISO date formats, they allow an optional zone ID on the end of the date (e.g. `+11:00`). I think given your findings an argument could be made for having separate `DateTimeFormatter`s for parsing and formatting. Use the lenient one for parsing and a more specific one for formatting.

Something else I noticed, if you set the zone (either through `timestampLongConverters.zoneId`) or through the Converter constructor, to something other than UTC the append performance is fast again. So perhaps the performance hit is only there in the case where you have not set the zone (or set it to UTC) and you're writing with the lenient formatter.

Do you want to fix this one, or do you want me to look at it? If you're doing it I think you should extend your perf test to include non-UTC zones to ensure we are creating the most efficient converter for each configuration.

**Comment by maxim-ponomarev** [comment]
@nicktindall I would really appreciate, if you could take a look.
Feel free to modify the test.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.24ea10](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.24ea10), [BOM-2.24ea43](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.24ea43)
